### PR TITLE
test: introduce nightly job for robustness test

### PR DESCRIPTION
**Pull request by fuweid**
* tests: Update TestRestartFromPowerFailure

Update case with a combination of EXT4 filesystem's commit setting and unexpected exit event. That EXT4 filesystem's commit is to sync all its data and metadata every seconds. The kernel can help us sync even if that process has been killed. With different commit setting, we can simulate that case that kernel syncs half part of dirty pages before power
failure. And for unexpected exit event, we can kill that process randomly or panic at failpoint instead of fixed code path.

* *: introduce nightly run for robustness test

Update GITHUB CI workflow for nightly run. Nightly run job takes 1h and is be verified in my fork https://github.com/fuweid/bbolt/actions/runs/7378167227/job/20072981571

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -84,4 +84,4 @@ test-failpoint:
 .PHONY: test-robustness # Running robustness tests requires root permission
 test-robustness:
 	go test -v ${TESTFLAGS} ./tests/dmflakey -test.root
-	go test -v ${TESTFLAGS} ./tests/robustness -test.root
+	go test -v ${TESTFLAGS} ${ROBUSTNESS_TESTFLAGS} ./tests/robustness -test.root
```
Comment (jmhbnz) [inline Makefile:87]: I notice we don't have any graceful validation that whoever runs `make test-robustness` has actually run the pre-requisite `make gofail-enable` prior.  Currently we just get cryptic error printed which folks might not understand at first glance, i.e:

```bash
=== RUN   TestRestartFromPowerFailure/fp_ext4_commit5s                                         
    powerfailure_test.go:112: start bbolt bench -work -path /tmp/TestRestartFromPowerFailurefp_ext4_commit5s797232614/002/boltdb -count=1000000000 -batch-size=5
    powerfailure_test.go:129: simulate power failure                                           
    powerfailure_test.go:134: random pick failpoint: resizeFileError                           
    powerfailure_test.go:168:                                                                  
                Error Trace:    /home/james/Documents/bbolt/tests/robustness/powerfailure_test.go:168
                                                        /home/james/Documents/bbolt/tests/robustness/powerfailure_test.go:135
                                                        /home/james/Documents/bbolt/tests/robustness/powerfailure_test.go:81
                Error:          Received unexpected error:                                     
                                Put "http://127.0.0.1:12345/resizeFileError": dial tcp 127.0.0.1:12345: connect: connection refused
```

In `etcd-io/etcd` we introduced https://github.com/etcd-io/etcd/blob/f8d5ba9a3fdff4f706f06ef0f0c0a27bb033ac53/tests/framework/integration/testing.go#L102. Perhaps we should introduce something similar here, not a blocker on this pr though.

**Review by jmhbnz — APPROVED** [review]
Overall LGTM thanks @fuweid.  Two suggestions that aren't blocking.

File: Makefile (hunk 1 above)
Comment (fuweid) [reply Makefile:87]: I build bbolt with failpoints. I think gofail.List doesn't work for this case. Let me think about how to handle.

File: tests/robustness/powerfailure_test.go [hunk 2]
```diff
@@ -23,9 +26,65 @@ import (
...
-	flakey := initFlakeyDevice(t, t.Name(), dmflakey.FSTypeEXT4, "")
+	for _, tc := range []struct {
+		name         string
+		du           time.Duration
+		fsMountOpt   string
+		useFailpoint bool
+	}{
+		{
+			name:         "fp_ext4_commit5s",
+			du:           5 * time.Second,
```
Comment (ahrtr) [inline tests/robustness/powerfailure_test.go:48]: It seems that the `du` is always 5 seconds?

**Comment by ahrtr** [comment]
Overall looks good to me. Please let me know if you want to resolve the comments in a separate PR, then I can merge this PR

**Comment by fuweid** [comment]
Hi @ahrtr , let me resolve those comments in a follow-up. Thanks

**Review by ahrtr — APPROVED** [review]
lgtm

Thanks

**Comment by fuweid** [comment]
It works https://github.com/etcd-io/bbolt/actions/workflows/robustness_nightly.yaml

**Pull request merged**